modules/core/src/main/scala/jsonrpclib/internals/RawMessage.scala

```diff
@@ -53,7 +53,37 @@ private[jsonrpclib] object RawMessage {
       RawMessage(`2.0`, result = Some(data.stripNull), id = Some(callId))
   }
 
-  implicit val rawMessageJsonValueCodecs: JsonValueCodec[RawMessage] =
-    JsonCodecMaker.make(CodecMakerConfig.withSkipNestedOptionValues(true))
+  // Custom encoder to flatten nested Option[Option[Payload]]
+  implicit val rawMessageEncoder: Encoder[RawMessage] = Encoder.instance { msg =>
+    Json
+      .obj(
+        "jsonrpc" -> Json.fromString(msg.jsonrpc),
+        "method" -> msg.method.asJson,
+        "params" -> msg.params.asJson,
+        "error" -> msg.error.asJson,
+        "id" -> msg.id.asJson
+      )
+      .deepMerge(
+        msg.result match {
+          case Some(Some(payload)) => Json.obj("result" -> payload.asJson)
+          case Some(None)          => Json.obj("result" -> Json.Null)
+          case None                => Json.obj()
+        }
+      )
+  }
```

Comment:
I don't like the deepMerge, it's not needed

```suggestion
  implicit val rawMessageEncoder: Encoder[RawMessage] = Encoder.instance { msg =>
    Json
      .obj(
        List(
          "jsonrpc" -> Json.fromString(msg.jsonrpc),
          "method" -> msg.method.asJson,
          "params" -> msg.params.asJson,
          "error" -> msg.error.asJson,
          "id" -> msg.id.asJson
        ) ++ {
          msg.result match {
            case Some(Some(payload)) => List("result" -> payload.asJson)
            case Some(None)          => List("result" -> Json.Null)
            case None                => Nil
          }
        }: _*
      )
  }
```